js/id/ui/alert.js

```diff
@@ -44,10 +44,8 @@ iD.ui.Alert = function(message,type,stack) {
 
     var prettyString = function(obj) {
         var finalString = '';
+        obj.commandDetail =  JSON.stringify(obj.commandDetail, null, '<br>');
```

Comment:
So without that if statement, it loops through and parses the JSON multiple times, making for a really ugly output (see below). The original code was how I got around that. If it weren't for the failing mocha tests, your simpler solution makes for better logic, not sure what the deal is.
![screenshot from 2017-05-04 14 59 04](https://cloud.githubusercontent.com/assets/17727812/25720485/109e5fe8-30db-11e7-83c8-f77386de3f3f.png)